### PR TITLE
Adding one more testdata cert-chain.  

### PR DESCRIPTION
**Pull request by Mercurrent**
Some tests need at least 2 differently rooted cert-chains.

### CODE REVIEW

File: trillian/testdata/Makefile [hunk 1]
```diff
@@ -111,6 +111,29 @@ distclean: clean
 	rm -f $(SERVER_PUBKEYS) fake-ca.cert int-ca.cert int-ca.csr.pem
 	rm -f $(LEAF_CERTS) $(LEAF_CSRS) $(LEAF_CHAINS)
 
+
+# We also use a second fake CA as a trust root for CT tests. This is its private key.
+CA_1_PRIVKEY=fake-ca1.privkey.pem
```
Comment (daviddrysdale) [inline trillian/testdata/Makefile]: nit: consistent basenames would be nice (`fake-ca1` vs `fake-ca-1`)

File: trillian/testdata/Makefile [hunk 2]
```diff
@@ -111,6 +111,29 @@ distclean: clean
...
+CA_1_PWD=ahenny
+
+# Corresponding Leaf certificates.
+LEAF_1_PRIVKEY=leaf-1.privkey.pem
```
Comment (daviddrysdale) [inline trillian/testdata/Makefile]: nit: consistent names preferred (`leaf-1` vs `subleaf-1`)

File: trillian/testdata/Makefile [hunk 3]
```diff
@@ -111,6 +111,29 @@ distclean: clean
...
+LEAF_1_PWD=louth
+
+# Fake Root CA 1
+fake-ca-1.cert: $(CA_1_PRIVKEY) fake-ca.cfg
+	openssl req -new -x509 -config fake-ca.cfg -set_serial 0x0406efac -days 3650 -extensions v3_ca1 -inform pem -key $(CA_1_PRIVKEY) -passin pass:$(CA_1_PWD) -out $@
```
Comment (daviddrysdale) [inline trillian/testdata/Makefile]: Can we have a different subject name for the new root cert, to reduce the chances of confusion

File: trillian/testdata/Makefile [hunk 4]
```diff
@@ -111,6 +111,29 @@ distclean: clean
...
+
+subleaf-1.csr.pem: $(LEAF_1_PRIVKEY)
+	openssl req -new -sha256 -key $(LEAF_1_PRIVKEY) -passin pass:$(LEAF_1_PWD) -subj "/C=GB/ST=London/O=Google/OU=Eng/CN=$@" -out $@
+subleaf-1.cert: subleaf-1.csr.pem fake-ca-1.cert
+	openssl x509 -req -in $< -sha256 -extfile fake-ca.cfg -extensions v3_user -CA fake-ca-1.cert -CAkey $(CA_1_PRIVKEY) -passin pass:$(CA_1_PWD) -set_serial 0xdeadbeaf -days 2600 -out $@
+
+# Short chain on CA 1
+subleaf1.chain: subleaf-1.cert fake-ca-1.cert
```
Comment (daviddrysdale) [inline trillian/testdata/Makefile]: nit: consistent names (`subleaf1` vs `subleaf-1`)

File: trillian/testdata/Makefile [hunk 5]
```diff
@@ -111,6 +111,29 @@ distclean: clean
...
+
+# Corresponding Leaf certificates.
+LEAF_1_PRIVKEY=leaf-1.privkey.pem
+LEAF_1_PWD=louth
+
+# Fake Root CA 1
+fake-ca-1.cert: $(CA_1_PRIVKEY) fake-ca.cfg
+	openssl req -new -x509 -config fake-ca.cfg -set_serial 0x0406efac -days 3650 -extensions v3_ca1 -inform pem -key $(CA_1_PRIVKEY) -passin pass:$(CA_1_PWD) -out $@
+
+subleaf-1.csr.pem: $(LEAF_1_PRIVKEY)
```
Comment (daviddrysdale) [inline trillian/testdata/Makefile]: What's **sub**leafy about this one?

File: trillian/testdata/Makefile (hunk 1 above)
Comment (Mercurrent) [reply trillian/testdata/Makefile]: Done

File: trillian/testdata/Makefile (hunk 2 above)
Comment (Mercurrent) [reply trillian/testdata/Makefile]: Done

File: trillian/testdata/Makefile (hunk 5 above)
Comment (Mercurrent) [reply trillian/testdata/Makefile]: Renamed to leaf.

File: trillian/testdata/Makefile (hunk 3 above)
Comment (Mercurrent) [reply trillian/testdata/Makefile]: What do you mean by subject name? ca-1 to ca-B or...?

File: trillian/testdata/Makefile (hunk 4 above)
Comment (Mercurrent) [reply trillian/testdata/Makefile]: Renamed.

**Comment by Mercurrent** [comment]
Renamed variables, PTAL

File: trillian/testdata/Makefile (hunk 3 above)
Comment (daviddrysdale) [reply trillian/testdata/Makefile]: I mean the subject name in the cert, which gets pulled from the `fake-ca.cfg` file; both `fake-ca.cert` and `fake-ca-1.cert` have `Subject: C=GB, O=Google, OU=Eng, L=London, ST=London, CN=FakeCertificateAuthority`

Comment (daviddrysdale) [reply trillian/testdata/Makefile]: It's hardly a big deal though - fine as is if that's easier.

**Comment by Mercurrent** [comment]
Thanks, David!

**Pull request merged**